flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SchedulerTestingUtils.java

```diff
@@ -213,4 +267,158 @@ public static CheckpointCoordinator getCheckpointCoordinator(SchedulerBase sched
 			return operatorGateway.sendOperatorEventToTask(task, operator, evt);
 		}
 	}
+
+	/**
+	 * Builder for {@link DefaultScheduler}.
+	 */
+	public static class DefaultSchedulerBuilder {
+		private final JobGraph jobGraph;
+
+		private SchedulingStrategyFactory schedulingStrategyFactory;
+
+		private Logger log = LOG;
+		private BackPressureStatsTracker backPressureStatsTracker = VoidBackPressureStatsTracker.INSTANCE;
+		private Executor ioExecutor = java.util.concurrent.Executors.newSingleThreadExecutor();
+		private Configuration jobMasterConfiguration = new Configuration();
+		private ScheduledExecutorService futureExecutor = new DirectScheduledExecutorService();
+		private ScheduledExecutor delayExecutor = new ScheduledExecutorServiceAdapter(futureExecutor);
+		private ClassLoader userCodeLoader = getClass().getClassLoader();
```

Comment:
`.getClassLoader()` can return `null`. I don't think this will happen in this code path but I would feel better if we used `ClassLoader.getSystemClassLoader()`